### PR TITLE
Applying consistant formatting across the project

### PR DESCRIPTION
**Pull request by whoisj**
Attempting to clean up and standardize the formatting of the code in the project.

Looks very disruptive, but contains no logic changes.

* White space usage
* Brace usage
* Indentation

### CODE REVIEW

**Comment by whoisj** [comment]
/CC @nulltoken 

(sorry this is so huge - I swear there are no logical changes)

File: LibGit2Sharp/RemoveFromIndexException.cs [hunk 1]
```diff
@@ -14,8 +14,7 @@ public class RemoveFromIndexException : LibGit2SharpException
         /// Initializes a new instance of the <see cref="UnmatchedPathException"/> class.
         /// </summary>
         public RemoveFromIndexException()
-        {
-        }
+        {        }
```
Comment (nulltoken) [inline LibGit2Sharp/RemoveFromIndexException.cs]: The spacing between the braces is a tad strange in this file.

**Comment by nulltoken** [comment]
@whoisj My head hurts a little bit, now :wink: 

Few questions:
 - How did you generate this? Through https://www.nuget.org/packages/CodeFormatter/ ? Would there be a way to add some .ps1 file that could be launched and automatically trigger the reformatting? (Note: Would that be possible, of course, it would be outside the scope of this PR)
 - Is there be a way to enforce a per line max length to avoid the horizontal scrolls when checking future diffs online?
 - Are you *really* ready to make every pending PR conflict and cope with the crowd hate? :wink:

**Comment by whoisj** [comment]
@nulltoken this was all done by hand, with loving care :heart:.

There's likely to be rather little PR collisions outside of a few files, and those collisions will be trivial to resolve (I've been testing against a few random PR which are pending).

So yes, I'm willing to help with confict resolves and manage crowd hate (as I think there will actually be very little). :pray:

**Comment by nulltoken** [comment]
>@nulltoken this was all done by hand, with loving care :heart:.

Wow! You're astonishing... 

We somehow really need to automate this kind of process, later.  /cc @shiftkey

File: LibGit2Sharp/RemoveFromIndexException.cs (hunk 1 above)
Comment (nulltoken) [inline LibGit2Sharp/RemoveFromIndexException.cs]: @whoisj Any chance you could fixup the commit? ^^

**Comment by shiftkey** [comment]
> We somehow really need to automate this kind of process, later.

I think it's worth looking at repurposing CodeFormatter to suit - I did this for one of my projects, and aside from the VS2015 requirement it's pretty great.

https://github.com/octokit/octokit.net/pull/807
https://github.com/shiftkey/codeformatter/pull/1

**Comment by whoisj** [comment]
There are always tools like [AStyle](http://astyle.sourceforge.net/) which are useful and do not require pre-release versions of Visual Studio to work. :smile:

**Comment by whoisj** [comment]
Is Travis on vacation today ? :tropical_drink:

**Comment by whoisj** [comment]
This is what Astyle produces:

https://github.com/libgit2/libgit2sharp/compare/vNext...whoisj:astyle

**Comment by whoisj** [comment]
@shiftkey I attempted to evaluate Octokit, but I can't even get the solution to load correctly let alone build or run. Could you set something up like I have for Astyle? :pray:

**Comment by whoisj** [comment]
This is what [Code Formatter](https://github.com/jaredpar/codeformatter) produces

https://github.com/libgit2/libgit2sharp/compare/vNext...whoisj:codeformatter

**Comment by shiftkey** [comment]
> I attempted to evaluate Octokit, but I can't even get the solution to load correctly let alone build or run. 

Could this be due to the VS2015 requirement? I've held off on merging this PR into Octokit proper until it RTMs because of this...

**Comment by whoisj** [comment]
> Could this be due to the VS2015 requirement? I've held off on merging this PR into Octokit proper until it RTMs because of this...

I'm using VS2015, but I get this fun error about needing to "click on the project and choose download", but that's not an option. :disappointed:

**Comment by shiftkey** [comment]
@whoisj weird, haven't seen that one before. could you open an issue over on the repo with some details so I can look into it when I get a chance?

**Comment by whoisj** [comment]
Given the difficulties Octokit is having due to VS migration, and the semi-ugliness Astyle creates, do we want to just merge this PR?

**Comment by Therzok** [comment]
Many PRs will become unmergeable, but sure.

/me silently curses this PR.

**Comment by nulltoken** [comment]
>Given the difficulties Octokit is having due to VS migration, and the semi-ugliness Astyle creates, do we want to just merge this PR?

@whoisj Could you fix in `RemoveFromIndexException` the spacing between braces?

**Comment by whoisj** [comment]
> @whoisj Could you fix in RemoveFromIndexException the spacing between braces?

@nulltoken done. :grinning:

**Comment by whoisj** [comment]
:tada: :balloon: :cake:

**Comment by nulltoken** [comment]
Thanks to you for having crafted this :heart:

**Pull request merged**